lib/src/repo.rs

```diff
@@ -843,40 +848,48 @@ impl MutableRepo {
     pub fn create_descendant_rebaser<'settings, 'repo>(
         &'repo mut self,
         settings: &'settings UserSettings,
+        options: RebaseOptions,
     ) -> DescendantRebaser<'settings, 'repo> {
         DescendantRebaser::new(
             settings,
             self,
             self.rewritten_commits.clone(),
             self.abandoned_commits.clone(),
+            options,
         )
     }
 
     fn rebase_descendants_return_rebaser<'settings, 'repo>(
         &'repo mut self,
         settings: &'settings UserSettings,
+        options: RebaseOptions,
     ) -> Result<Option<DescendantRebaser<'settings, 'repo>>, TreeMergeError> {
         if !self.has_rewrites() {
             // Optimization
             return Ok(None);
         }
-        let mut rebaser = self.create_descendant_rebaser(settings);
+        let mut rebaser = self.create_descendant_rebaser(settings, options);
         rebaser.rebase_all()?;
         Ok(Some(rebaser))
     }
 
-    pub fn rebase_descendants(&mut self, settings: &UserSettings) -> Result<usize, TreeMergeError> {
+    pub fn rebase_descendants(
+        &mut self,
+        settings: &UserSettings,
+        options: RebaseOptions,
+    ) -> Result<usize, TreeMergeError> {
```

Comment:
Since almost all callers want to keep empty commits, maybe it's better to keep the existing function with the existing behavior and add a new function taking the `RebaseOptions`?